### ISSUE TITLE
getInvoiceAsPdf returns XML

### ISSUE THREAD

**Issue by DrewynAppscore** [XeroAPI/xero-php-oauth2#60]
Xero's API requires an Accept header with 'application/pdf' to receive back a PDF version of the invoice. 

Currently the `getInvoiceAsPdfRequest` function has a lot of bloated code that does nothing (useless if statements, etc.)  and `headerSelector->selectHeaders` is setting the Accept header to 'application/octet-stream' which is causing Xero to return the default XML structure of the Invoice.

I'm also not sure why it requires a Content-Type parameter because it should be only returning a PDF.

**Comment by manigandanta** [comment]
I am also facing the same issue, Will this issue be fixed soon? Is there any workaround for this.

**Comment by SidneyAllen** [comment]
@DrewynAppscore and @manigandanta - confirmed this issue and we'll be looking into it.

**Comment by IanSimpson** [comment]
Also an issue here. Worth mentioning that `getCreditNoteAsPdf` function is broken in the same way, so any fix will need to cover both.

I've put a workaround in place, though it's not a good solution as it'll be overwritten by updates. In AccountingApi.php, in the `getInvoiceAsPdfWithHttpInfo` and `getCreditNoteAsPdfWithHttpInfo` find the `$options = $this->createHttpClientOption();` line (right near the top of the function), and after it add:

```
$options['headers'] = [
    'Accept' => 'application/pdf',
];
```

Not a fix, but allows me to continue with my migration.

**Comment by SidneyAllen** [comment]
@DrewynAppscore and @manigandanta - this issue is fixed in version 1.2.1 and higher for both Invoices and CreditNotes

Code I used to test it .... use as you see fit :-)

$invoices = $apiInstance->getInvoices($xeroTenantId); 						
$invoiceId = $invoices->getInvoices()[0]->getInvoiceId();
$result = $apiInstance->getInvoiceAsPdf($xeroTenantId, $invoiceId, "application/pdf"); 						

// read PDF contents
$content = $result->fread($result->getSize());

//check if a temp dir exsits
$dir_to_save = "./temp/";
if (!is_dir($dir_to_save)) {
  mkdir($dir_to_save);
}
// write to temp dir
file_put_contents($dir_to_save . $result->getFileName() . ".pdf", $content);
